application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -536,6 +550,39 @@ type FeedPublisher {
   url: String!
 }
 
+"""
+Exact dated stoptime represents the time when a specific trip on a specific date arrives to and/or departs from a specific stop.
+This can include realtime estimates.
+"""
+type FixedStopTimeOnServiceDate {
+  "Scheduled arrival time to the stop and a realtime estimate, if one exists."
+  arrival: ArrivalDepartureTime
+  "Scheduled departure time from the stop and a realtime estimate, if one exists."
+  departure: ArrivalDepartureTime
+  "The stop where this arrival/departure happens"
+  stop: Stop
+}
+
+"A fixed (i.e. not flexible or frequency based) trip on a specific service date"
+type FixedTripOnServiceDate {
+  """
+  The service date when the trip occurs.
+    
+  **Note**: A service date is a technical term useful for transit planning purposes and might not
+    correspond to a how a passenger thinks of a calendar date. For example, a night bus running
+    on Sunday morning at 1am to 3am, might have the previous Saturday's service date.
+  """
+  date: LocalDate!
+  "The time when the trip ends at a stop and information related to the stop."
+  end: FixedStopTimeOnServiceDate
+  "The time when the trip starts from a stop and information related to the stop."
```

Comment:
What happens when first or last are cancelled? Maybe modify the language to "scheduled to start ...".

Reply:
If you just want to tell that the departure of X route from A to B is between 14:10-16:20 is cancelled, you don't have to fetch all the stop times. We discussed the possibility to get nth stop time but that doesn't really work when you fetch a list of trips since they can have random number of stops.